sklearn/decomposition/_nmf.py

```diff
@@ -705,20 +772,31 @@ def _multiplicative_update_h(X, W, H, beta_loss, l1_reg_H, l2_reg_H, gamma):
         denominator = denominator + l2_reg_H * H
     denominator[denominator == 0] = EPSILON
 
-    numerator /= denominator
-    delta_H = numerator
+    if A is not None and B is not None:
+        A *= rho
+        B *= rho
+        A += numerator * H**2
+        B += denominator
+        numerator = A
+        denominator = B
+        H = (np.divide(A, B))**0.5
+    else:
+        numerator /= denominator
+        delta_H = numerator
 
-    # gamma is in ]0, 1]
-    if gamma != 1:
-        delta_H **= gamma
+        # gamma is in ]0, 1]
+        if gamma != 1:
+            delta_H **= gamma
+        H = delta_H * H_old
 
-    return delta_H
+    return H, A, B
```

Comment:
Your comment made me check again on A and B: from Lefevre-Bach-Fevotte it seems to me that A and N are initialized at each iteration (outside the batch loop). I have rewritten the code so A and B are no longer called outside `_multiplicative_update_h`. Is my interpretation correct?

Reply:
No, A and B should be initialized once at the beginning and then updated at each iteration. Maybe it will make things clearer and more similar to other online algos in sklearn to change the nested loop
```
for i in range(max_iter):
    for batch in X
```
into a single loop
```
batches = gen_batches(n_samples, self._batch_size)
batches = itertools.cycle(batches)

n_steps_per_epoch = int(np.ceil(n_samples / self._batch_size))
n_steps = self.max_iter * n_steps_per_epoch

for i, batch in zip(range(n_steps), batches):
```

Reply:
Ok, I will try that. Thanks!